pkg/profiler/cpu/cpu.go

```diff
@@ -849,7 +851,7 @@ func (p *CPU) obtainProfiles(ctx context.Context) ([]*profiler.Profile, error) {
 				locationIndex, ok := locationIndices[id][addr]
 				if !ok {
 					locationIndex = len(locations[id])
```

Comment:
All this tracking of location indexes is exactly about the deduplication and monotonic increasing IDs.